### PR TITLE
v0.31.5 changelog and version updates

### PR DESCRIPTION
**Pull request by melekes**
https://github.com/tendermint/tendermint/blob/4474a5ec70fad39d78b28a543d72c25715bfca84/CHANGELOG.md#v0315

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,28 @@
...
+This release fixes a regression from v0.31.4 where, in existing chains that
+were upgraded, `/validators` could return an empty validator set. This is true
+for almost all heights given the validator set remains the same.
+
+Special thanks to external contributors on this release:
+@brapse, @guagualvcha, @dongsam, @phucc
+
+### IMPROVEMENTS:
+
+- [libs/common] CMap: slight optimization in Keys() and Values() (@phucc)
```
Comment (Liamsi) [inline CHANGELOG.md]: Nit:
```suggestion
- [libs/common] `CMap`: slight optimization in `Keys()` and `Values()` (@phucc)
```
(We do not consistently mark code / symbols as such but I think we should).

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,28 @@
 # Changelog
 
+## v0.31.5
+
+*April 16th, 2019*
+
+This release fixes a regression from v0.31.4 where, in existing chains that
+were upgraded, `/validators` could return an empty validator set. This is true
+for almost all heights given the validator set remains the same.
```
Comment (Liamsi) [inline CHANGELOG.md]:
```suggestion
for almost all heights, given the validator set remains the same.
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,28 @@
 # Changelog
 
+## v0.31.5
+
+*April 16th, 2019*
+
+This release fixes a regression from v0.31.4 where, in existing chains that
+were upgraded, `/validators` could return an empty validator set. This is true
```
Comment (Liamsi) [inline CHANGELOG.md]: Suggestion: 
"This release fixes a regression from v0.31.4 where, in existing chains that
were upgraded, querying `/validators` via RPC could return an empty validator set. 
This happened for almost all height, given the validator set remained the same."
?

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,5 +1,28 @@
...
+
+### IMPROVEMENTS:
+
+- [libs/common] CMap: slight optimization in Keys() and Values() (@phucc)
+- [gitignore] gitignore: add .vendor-new (@dongsam)
+
+### BUG FIXES:
+
+- [state] [\#3537](https://github.com/tendermint/tendermint/pull/3537#issuecomment-482711833)
+  LoadValidators: do not return an empty validator set
```
Comment (Liamsi) [inline CHANGELOG.md]:
```suggestion
  `LoadValidators`: do not return an empty validator set
```

File: CHANGELOG.md (hunk 3 above)
Comment (melekes) [reply CHANGELOG.md]: I like my version better :) I'll let @ebuchman be the judge

Comment (ebuchman) [reply CHANGELOG.md]: Ismail's is defntly clearer that this is about the RPC endpoint, though we often refer to `/xxx` and expect the reader to know we're talking about the RPC.

**Pull request merged**